packages/flame_rive/lib/src/rive_component.dart

```diff
@@ -156,12 +156,23 @@ class RiveArtboardRenderer {
   }
 }
 
+/// loads the Artboard from the specified Rive File
+///
+/// when [artboardName] is not null, Returns an artboard from the specified
+/// name, or [AssertionError] if no artboard with that name exists in the file
 Future<Artboard> loadArtboard(
   FutureOr<RiveFile> file, {
   String? artboardName,
 }) async {
   final loaded = await file;
-  return artboardName == null
-      ? loaded.mainArtboard.instance()
-      : loaded.artboardByName(artboardName)!.instance();
+  if (artboardName == null) {
+    return loaded.mainArtboard.instance();
+  } else {
+    final artboard = loaded.artboardByName(artboardName)?.instance();
+    assert(
+      artboard != null,
+      'No artboard with the specified name exists in the RiveFile',
+    );
+    return artboard!;
```

Comment:
This line will blow up if the assertion is triggered, is that the point? Or should it maybe return the main artboard if it is null?

Reply:
yep, This line doesn't execute when assertion is triggered. 
Assertion is triggered only in debug mode, It ignores in the Production. maybe we can return main Artboard for this?

Reply:
Maybe it is good to fail hard here anyways, let's keep it like it is